GW2EIEvtcParser/EncounterLogic/Fractals/Nightmare/Siax.cs

```diff
@@ -99,5 +100,148 @@ internal override List<PhaseData> GetPhases(ParsedEvtcLog log, bool requirePhase
             return phases;
         }
 
+        internal override void ComputeNPCCombatReplayActors(NPC target, ParsedEvtcLog log, CombatReplay replay)
+        {
+            IReadOnlyList<AbstractCastEvent> casts = target.GetCastEvents(log, log.FightData.FightStart, log.FightData.FightEnd);
+
+            switch (target.ID)
+            {
+                case (int)ArcDPSEnums.TargetID.Siax:
+                    // Siax's Breakbar
+                    var causticExplosionBreakbar = casts.Where(x => x.SkillId == CausticExplosionSiaxBreakbar).ToList();
+                    foreach (AbstractCastEvent c in causticExplosionBreakbar)
+                    {
+                        int duration = 15000;
+                        int start = (int)c.Time;
+                        int expectedHitTime = (int)c.Time + duration;
+                        int attackEnd = (int)c.Time + duration;
+                        int stunTime = attackEnd;
+                        int detTime = attackEnd;
+
+                        IReadOnlyDictionary<long, BuffsGraphModel> bgmUptime = target.GetBuffGraphs(log);
+
+                        BuffsGraphModel bgmStun = bgmUptime.TryGetValue(Stun, out BuffsGraphModel value) ? value : null;
+                        if (bgmStun != null)
+                        {
+                            Segment segment = bgmStun.BuffChart.FirstOrDefault(x => x.Start > c.Time && x.Start < c.Time + duration);
```

Comment:
You should be checking for x.Value, the segment returned could very well a period of time where the target was not stunned. Same for the others below. You can also use the "GetBuffStatus" API on target to reduce quite a bit of code regarding stun checking.